src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -185,6 +185,7 @@ public RubyContext(RubyLanguage language, TruffleLanguage.Env env) {
 
         Launcher.printTruffleTimeMetric("before-thread-manager");
         threadManager = new ThreadManager(this);
+        threadManager.initialize();
```

Comment:
Unfortunately it would not solve the problem here, we need the assignment in RubyContext to happen so methods called by `initialize()` can use `context.getThreadManager()`.